### PR TITLE
Support DATE_FANCINESS and updated in base and base-jinja

### PR DESCRIPTION
**Pull request by gwax**
### Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [x] I tested my changes.

### Description

This PR changes two things relating to base/base-jinja handling of datetime in posts:

1. addition of "updated" time, if specified in the post metadata (output is unchanged if "updated" no specified)
1. DATE_FANCINESS is supported for published and updated dates (output is unchanged if ``DATE_FANCINESS=0``

### CODE REVIEW

**Comment by gwax** [comment]
PR fails on baseline check (because of changed content) but is otherwise passing.

I have tested the changes locally against my own nikola setup.

File: CHANGES.txt [hunk 1]
```diff
@@ -8,6 +8,14 @@ Features
 
 * Don’t generate gallery index if the destination directory is
   site root and it would conflict with blog index (Issue #3133)
+* ``base`` and ``base-jinja`` templates now support updated
```
Comment (Kwpolska) [inline CHANGES.txt]: Please copy the change to `bootblog4/index.tmpl` and change the line to

    * All built-in themes now support ``updated``

File: nikola/data/themes/base-jinja/templates/index.tmpl [hunk 2]
```diff
@@ -37,7 +37,15 @@
                 {{ post.author()|e }}
             {% endif %}
             </span></p>
-            <p class="dateline"><a href="{{ post.permalink() }}" rel="bookmark"><time class="published dt-published" datetime="{{ post.formatted_date('webiso') }}" title="{{ post.formatted_date(date_format)|e }}">{{ post.formatted_date(date_format)|e }}</time></a></p>
+            <p class="dateline">
+            <a href="{{ post.permalink() }}" rel="bookmark">
+            <time class="published dt-published" datetime="{{ post.formatted_date('webiso') }}" itemprop="datePublished" title="{{ post.formatted_date(date_format)|e }}">{{ post.formatted_date(date_format)|e }}</time>
+            {% if post.updated and post.updated != post.date %}
+                <span class="updated"> updated </span>
```
[the remark this answers is not in the document]
Comment (gwax) [reply nikola/data/themes/base-jinja/templates/index.tmpl]: I'm not quite sure how to have the messages update span across the `<time>...</time>` section but I can certain create a message around "updated".

**Comment by gwax** [comment]
Done.

I know that updating the messages/* files by hand is not the right way to do things and liable to break the next time someone updates the translations but I could not figure out how to request access to Transifex as an English speaker or otherwise add a new message type.

File: nikola/data/themes/base-jinja/templates/index.tmpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kwpolska) [reply nikola/data/themes/base-jinja/templates/index.tmpl]: Come to think about it, it won’t be easy to do. Let’s scrap that.

**Comment by Kwpolska** [comment]
Regarding translations, adding them to Transifex requires admin rights, I’ll take care of that.

**Comment by gwax** [comment]
@Kwpolska changing to (updated XX) is totally fine in my book but I believe you forgot to run `scripts/jingify.py` because the jinja versions do not appear to have been updated.

Thanks.

**Comment by Kwpolska** [comment]
Yeah, I forgot. Fixed on master.

**Pull request merged**